website/status.php

```diff
@@ -1,37 +1,57 @@
 <?php
+
 @define('CONST_ConnectionBucket_PageType', 'Status');
 
 require_once(dirname(dirname(__FILE__)).'/settings/settings.php');
 require_once(CONST_BasePath.'/lib/init-website.php');
+require_once(CONST_BasePath.'/lib/ParameterParser.php');
+require_once(CONST_BasePath.'/lib/Status.php');
 
-
-function statusError($sMsg)
-{
-    header('HTTP/1.0 500 Internal Server Error');
-    echo 'ERROR: '.$sMsg;
-    exit;
-}
+$oParams = new Nominatim\ParameterParser();
+$sOutputFormat = $oParams->getSet('format', array('text', 'json'), 'text');
 
 $oDB =& DB::connect(CONST_Database_DSN, false);
-if (!$oDB || PEAR::isError($oDB)) {
-    statusError('No database');
-}
+$oStatus = new Nominatim\Status($oDB);
+
+$sStatus = $oStatus->status();
+$bGotError = ($sStatus != 'OK');
```

Comment:
Having to parse a string for an error is not particularly efficient. How about having status() return the same array as you set up in $aResponse below, i.e. with status, code and message?

Reply:
I cleaned that up a little but didn't want to put HTTP error codes into the lib/ to separate logic from presentation. Previously I had the `status` method throw exceptions, it made tests a bit unreadable. Could add that again.